File: features/step_definitions/jenkins_configuration_steps.rb
#!/usr/bin/env ruby

Given /^I add Java version "([^"]*)" with name "([^"]*)" installed automatically to Jenkins config page$/ do |version, name|
  @runner.wait_for_updates 'JDK'
  @jenkins_config = Jenkins::JenkinsConfig.get(@base_url, 'Jenkins global configuration')
  @jenkins_config.enter_oracle_credentials(ENV['ORACLE_LOGIN'],ENV['ORACLE_PASSWORD'])
  @jenkins_config.configure do
    @jenkins_config.add_tool("JDK")
    @jenkins_config.add_jdk_auto_installation(name,version)
  end
end

Then /^I should be able to configure (.*) globally$/ do |section_name|
  @jenkins_config = Jenkins::JenkinsConfig.get(@base_url, 'Jenkins global configuration')
  @jenkins_config.open
  page.should have_xpath "//div[@class='section-header' and normalize-space(text())='#{section_name}']"
end

File: src/main/java/org/jenkinsci/test/acceptance/steps/JenkinsConfigurationSteps.java
package org.jenkinsci.test.acceptance.steps;

import cucumber.api.java.en.Given;
import org.jenkinsci.test.acceptance.controller.JenkinsController;
import org.jenkinsci.test.acceptance.po.JenkinsConfig;

import javax.inject.Inject;

/**
 * @author Kohsuke Kawaguchi
 */
public class JenkinsConfigurationSteps extends AbstractSteps {
    @Inject
    JenkinsController controller;

    @Given("^I add Java version \"([^\"]*)\" with name \"([^\"]*)\" installed automatically to Jenkins config page$")
    public void I_add_Java_version_with_name_installed_automatically_to_Jenkins_config_page(String version, String name) throws Throwable {
        controller.waitForUpdates();
        JenkinsConfig c = my.jenkins.getConfigPage();
        c.enterOracleCredential(System.getenv("ORACLE_LOGIN"), System.getenv("ORACLE_PASSWORD"));
        c.configure();
        c.addTool("JDK");
        c.addJdkAutoInstallation(name,version);
        c.save();
    }
}
